src/custom/state/affiliate/updater.tsx

```diff
@@ -0,0 +1,20 @@
+import { useEffect } from 'react'
+
+import { updateAffiliateLink } from '@src/custom/state/affiliate/actions'
+import useReferralLink from '../../hooks/useReferralLink'
+import { useWalletInfo } from '../../hooks/useWalletInfo'
+import { useAppDispatch } from '@src/state/hooks'
+
+export default function ReferralLinkUpdater() {
+  const dispatch = useAppDispatch()
+  const referralLink = useReferralLink()
```

Comment:
Actually, this returns your own affiliate link, the one you want to share with other people.
What we want here is to store what's coming from the url param when someone else opens your URL affiliate link.

You'll want to get the value Ramiro is fetching on this file from the PR https://github.com/gnosis/cowswap/pull/953/files#diff-af06422a6866f6f150604ce5491f449819b0882344b8da8b2387f6288fe74949L1

See the comments there.
The one where you said useless code? That's where you'll trigger your stuff :)

Reply:
> What we want here is to store what's coming from the url param when someone else opens your URL affiliate link.

@alfetopito I believe there's a misunderstanding with the ticket's description:

> Persist in the global state of the app, associated with the connected address, remember the user can change addresses any time

The ticket mentions the connected address while your comment says its the address from the url.